docs/src/main/mdoc/datatypes/statet.md

```diff
@@ -0,0 +1,244 @@
+---
+layout: docs
+title:  "StateT"
+section: "data"
+source: "core/src/main/scala/cats/data/package.scala"
+scaladoc: "#cats.data.StateT"
+---
+# StateT
+
+`StateT[F[_], S, A]` is a data type used to define state machine
+effectful computations that can be easily composed into a larger one.
+
+The type parameters are:
+- `F[_]` represents the effect in which the computation is performed.
+- `S` represents the underlying state, shared between each step of the
+  state machine.
+- `A` represents the return value.
+
+It can be seen as a way to add the capability to manipulate a shared
+state, to an existing computation in the context of `F`.
+
+This definition could be confusing, but it will become clear after
+learning the `State` monad and by the example below.
+
+## StateT and State Relationship
+
+`StateT` is a monad transformer for
+[State](https://typelevel.org/cats/datatypes/state.html), in
+particular [State](https://typelevel.org/cats/datatypes/state.html) is
+defined as a `StateT` with
+[Eval](https://typelevel.org/cats/datatypes/eval.html) as the effect
+`F`.  Therefore, `StateT` exposes the same methods of
+[State](https://typelevel.org/cats/datatypes/state.html), such as:
+`modify`, `get` and `set`.  Plus additional methods, that handles
+effectful computations, eg: `modifyF`, `setF` and `liftF`.
```

Comment:
Oh, I see you already have this here :) 👍 IMO we should have this at the beginning. I think it's difficult to understand `StateT` without understanding `State`.

Reply:
I agree about the fact that it's hard to understand `StateT` without knowing `State`. My idea was to give an initial brief general definition of the `StateT` datatype and then specify the relationship with `State`. At the end of the day, it's just <20 lines below 😅 

I can definitely add the type synonym for `State` in here for clarity.

Reply:
I agree, a brief general definition is always very good :) "state machine effectful computations" is something that's really hard to know what that means though, IMO. Even "state machiner" and "effect" by themselves are kind of complex!

Reply:
True, the problem is that some stuff is not at ground level by definition. 
What we can do is to put reference links where the reader can go and grasp the "prerequisite concepts", eg a Wikipedia link to [state machine](https://en.wikipedia.org/wiki/Finite-state_machine) or a youtube link to what an effect system is (I just attended an amazing one as keynote in ScalaLove 2022 <3 )